### ISSUE TITLE
Fix S1144 FP: Private fields in types marked with the StructLayout attribute

### ISSUE THREAD

**Issue by zsolt-kolbay-sonarsource** [SonarSource/sonar-dotnet#6868]
### Description

The [StructLayout attribute](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.structlayoutattribute) is used to control the exact memory layout of class/struct when passing instances of it to unmanaged code.
In cases like that, unreferenced private fields are sometimes used as fillers, referencing real data, that is never used by the code, but must still be part of the class/struct in order to have a proper layout in the memory.

### Repro steps

```
[StructLayout(LayoutKind.Sequential)]
public ref struct UnmanagedType
{
    private IntPtr ArgCookie; // Noncompliant - FP: unused private field in a type marked with StructLayout should not raise an issue
}
```

A more detailed sample:
https://peach.sonarsource.com/issues?projects=dotnet-runtime&resolved=false&rules=csharpsquid%3AS1144&open=AX0U1N1GAng_OcXOZ5RD

### Expected behavior

No issue should be raised.

### Actual behavior

An issue is raised: Remove the unused private field 'xxxxx'.

### Known workarounds

Making the variable non-`private` stops raising the issue.

### Related information

* C#/VB.NET 8.53

**Comment by martin-strecker-sonarsource** [comment]
Duplicate of https://github.com/SonarSource/sonar-dotnet/issues/6912
